### DISCUSSION TITLE
Wall of misunderstanding layers and modules

### DISCUSSION THREAD

**Discussion by alphaCastor** [nuxt/nuxt#18493] [Questions]
Hi there. I am trying to understand how to architecture rich functionality project. It has a lot of features. Firstly, before layers where introduced, I thought it can be done by modules.

So for each feature I created separate module:

 - forum feature is a separate module with all related business logic and components. Pages and components are extended to app, so I can use its components where I want
 - video game feature. The same as I have at forum but related with that feature
 - etc...
 
Also I had some situation when one module needs some components from another. In that case I just used that module in playground and it was ok. 
 
 For now now layers seems more suitable for such case. Can we use it for feature based architecture and not only for some custom shared components, configs, .etc.?
 
 Can you please share your opinion on when is better to use module and when layers (in addition to documentation)

**Comment by OlaAlsaker** [comment]
Nuxt has provided some documentation about layers:
https://nuxt.com/docs/getting-started/layers

.. and here:
https://nuxt.com/docs/guide/going-further/layers

You can also follow the work on the layers-functionality in #13367
